### ISSUE TITLE
User specific catalog

### ISSUE THREAD

**Issue by adriengoeller** [phlowers/mechaphlowers#47]
### Describe the current behavior

to do after #46  

### Describe the expected behavior

The user must be able to indicate where his data is located and load it. The load function then returns an instantiation of the class that will be the library with its adapter.

The user uses a load function in the data module. A generic function will read the configuration yaml. It will select the catalog type to be loaded. If the type is not available, an error will occur.
The yaml will give the location of the data and allow you to set the expected columns.

```yaml
Name: Pokemon
version: 0.1
dir: ".../.../path/to/csv"
catalog_type: FakeCatalog
```

**Comment by adriengoeller** [comment]
TBD: take the "at import" part in another issue

**Comment by ai-qui** [comment]
I've looked for python libraries to parse yaml files and found [ruamel.yaml](https://pypi.org/project/ruamel.yaml/), which is compatible with pyodide (cf. the [list of compatible packages](https://pyodide.org/en/0.25.0/usage/packages-in-pyodide.html)).

But the documentation for ruamel.yaml states that future uploads to pypi may become impossible, so I wonder if we should use it 🤔 

I also came accross [pyyaml](https://pypi.org/project/PyYAML/), which doesn't seem to be directly supported by pyodide, but may be available in a pyodide-compatible version on [anaconda's repository](https://anaconda.org/pyodide/pyyaml).

To be continued...

**Comment by adriengoeller** [comment]
solved by #258
